dashboard/src/components/DeploymentForm/DeploymentForm.tsx

```diff
@@ -85,14 +89,19 @@ class DeploymentForm extends React.Component<IDeploymentFormProps, IDeploymentFo
 
     if (nextProps.selected.version && nextProps.selected.version !== this.props.selected.version) {
       getChartValues(chartID, nextProps.selected.version.attributes.version);
+      getChartSchema(chartID, nextProps.selected.version.attributes.version);
       return;
     }
 
     if (!this.state.valuesModified) {
       if (version) {
-        this.setState({ appValues: nextProps.selected.values });
+        this.setState({ appValues: nextProps.selected.values || "" });
       }
     }
+
+    if (nextProps.selected.schema) {
+      this.setState({ basicFormParameters: retrieveBasicFormParams(nextProps.selected.schema) });
+    }
```

Comment:
What version of react are we using? This should be done in https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops , as componentWillReceiveProps is being removed.

Reply:
We are using `^16.2.0` but yes, we will need to update that hook (we are using it in many places). I will create a separate issue for that.

Reply:
https://github.com/kubeapps/kubeapps/issues/1200